File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F07.P2.go
package main

import "fmt"

func probe() (int32, int32, float64) {
	var a int32 = -7
	var b int32 = 2
	// BEGIN PROBE F07.P2
	q := a / b
	m := a % b
	d := float64(a) / float64(b)
	// END PROBE F07.P2
	return q, m, d
}

func main() {
	fmt.Println(probe())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F07.P2.swift
func probe(_ a: Int32, _ b: Int32) -> (Int32, Int32, Double) {
// BEGIN PROBE F07.P2
    let q = a / b
    let m = a % b
    let d = Double(a) / Double(b)
// END PROBE F07.P2
    return (q, m, d)
}

let out = probe(-7, 2)
print(out.0, out.1, out.2)
